protocol/x/clob/abci.go

```diff
@@ -169,7 +169,26 @@ func PrepareCheckState(
 	// For orders that are filled in the last block, send an orderbook update to the grpc streams.
 	if keeper.GetGrpcStreamingManager().Enabled() {
 		allUpdates := types.NewOffchainUpdates()
+		orderIdsToSend := make(map[types.OrderId]bool)
+
+		// Send an update for reverted local operations.
+		for _, operation := range localValidatorOperationsQueue {
+			if match := operation.GetMatch(); match != nil {
+				orderIdsToSend[match.GetMatchOrders().TakerOrderId] = true
+
+				for _, fill := range match.GetMatchOrders().Fills {
+					orderIdsToSend[fill.MakerOrderId] = true
+				}
+			}
+		}
+
+		// Send an update for orders that were proposed.
 		for _, orderId := range processProposerMatchesEvents.OrderIdsFilledInLastBlock {
+			orderIdsToSend[orderId] = true
```

Comment:
just double checking when we replay the orders we will send `keeper.MemClob.GetOrderbookUpdatesForOrderUpdate(ctx, orderId)` during the regular place order flow right?

Reply:
yup - 

for orders in `processProposerMatchesEvents.OrderIdsFilledInLastBlock`, the intention is to send an update for orders that are included by BP but not in local queue.

for orders in local operations queue, the intention is to send an update for orders that are still on the book ([link](https://github.com/dydxprotocol/v4-chain/blob/main/protocol/x/clob/memclob/memclob.go#L1176-L1181)). replay would fail b/c the order already exists, so no updates are sent.